test/new-e2e/pkg/utils/e2e/client/agent_commands.go

```diff
@@ -6,152 +6,157 @@
 package client
 
 import (
-	"errors"
+	"fmt"
 	"strings"
 	"testing"
 	"time"
 
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client/agentclient"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/optional"
+
 	"github.com/cenkalti/backoff"
 	"github.com/stretchr/testify/require"
 )
 
-// executeAgentCmdWithError is a function to run an Agent command.
-type executeAgentCmdWithError func(arguments []string) (string, error)
+type agentCommandExecutor interface {
+	execute(arguments []string) (string, error)
+}
 
-// agentCommandRunner is an internal type that provides methods to run Agent commands.
+// agentCommands is an internal type that provides methods to run Agent commands.
 // It is used by both [VMClient] and [Docker]
-type agentCommandRunner struct {
-	t                        *testing.T
-	executeAgentCmdWithError executeAgentCmdWithError
-	isReady                  bool
+type agentCommands struct {
+	t        *testing.T
+	executor agentCommandExecutor
+	isReady  bool
 }
 
 // Create a new instance of agentCommandRunner
-func newAgentCommandRunner(t *testing.T, executeAgentCmdWithError executeAgentCmdWithError) *agentCommandRunner {
-	agent := &agentCommandRunner{
-		t:                        t,
-		executeAgentCmdWithError: executeAgentCmdWithError,
-		isReady:                  false,
+func newAgentCommandRunner(t *testing.T, executor agentCommandExecutor) *agentCommands {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
I preferred `agentCommandRunner`, more explicitly states that can run commands, rather than return command strings